### PR TITLE
Disallow binning in inner dimensions (temporary)

### PR DESCRIPTION
**Pull request by jl-wynen**
This is a temporary workaround for #1479 so that users don't get infinite loops but a somewhat informative error message.

### CODE REVIEW

File: variable/include/scipp/variable/bucket_model.h [hunk 1]
```diff
@@ -126,6 +126,16 @@ template <class T> class DataModel<bucket<T>> : public VariableConcept {
             return x.second >= 0 && x.first > x.second;
           }) != vals.end())
         throw except::SliceError("Overlapping bucket indices are not allowed.");
+      if (buffer.dims().ndim() > 1 && buffer.dims().index(dim) != 0) {
+        if (std::find_if(vals.begin(), vals.end(), [](const auto x) {
+              return x.first == x.second;
+            }) != vals.end()) {
```
Comment (SimonHeybrock) [inline variable/include/scipp/variable/bucket_model.h:132]: I think we should remove this check: Bin sizes can be changed *after* creation without checks, so we should throw here even if there are not empty buckets?

Comment (jl-wynen) [reply variable/include/scipp/variable/bucket_model.h:132]: That would be very conservative. But then we would be on the safe side.

Comment (jl-wynen) [reply variable/include/scipp/variable/bucket_model.h:132]: Actually, the error message doesn't say anything about empty bins...

**Review by SimonHeybrock — APPROVED** [review]

**Comment by SimonHeybrock** [comment]
I believe you said this will be handled as part of #1483, closing.

**Pull request closed without merging**